python-sdk/nuimages/nuimages.py

```diff
@@ -238,108 +298,521 @@ def list_categories(self, image_tokens: List[str] = None) -> None:
+            Set to any other value to rotate relative to the driving direction (in radians).
+        :param center_key_pose: Whether to center the trajectory on the key pose.
+        :return: (
+            translations: A matrix with sample_datas x 3 values of the translations at each timestamp.
+            key_index: The index of the translations corresponding to the keyframe (usually 6).
+        )
+        """
+        # Get trajectory data.
+        timestamps, translations = self.get_ego_pose_data(sample_token)
+
+        # Find keyframe translation and rotation.
+        sample = self.get('sample', sample_token)
+        sd_camera = self.get('sample_data', sample['key_camera_token'])
+        ego_pose = self.get('ego_pose', sd_camera['ego_pose_token'])
+        key_rotation = Quaternion(ego_pose['rotation'])
+        key_timestamp = ego_pose['timestamp']
+        key_index = [i for i, t in enumerate(timestamps) if t == key_timestamp][0]
+
+        # Rotate points such that the initial driving direction points upwards.
+        if rotation_yaw is not None:
+            rotation = key_rotation.inverse * Quaternion(axis=[0, 0, 1], angle=np.pi / 2 - rotation_yaw)
+            translations = np.dot(rotation.rotation_matrix, translations.T).T
+
+        # Subtract origin to have lower numbers on the axes.
+        if center_key_pose:
+            translations -= translations[key_index, :]
+
+        return translations, key_index
+
+    # ### Rendering methods. ###
 
     def render_image(self,
-                     image_token: str,
+                     sd_token_camera: str,
                      with_annotations: bool = True,
+                     with_category: bool = False,
                      with_attributes: bool = False,
-                     box_tokens: List[str] = None,
+                     object_tokens: List[str] = None,
                      surface_tokens: List[str] = None,
-                     render_scale: float = 2.0,
-                     ax: Axes = None) -> PIL.Image:
+                     render_scale: float = 1.0,
+                     out_path: str = None) -> None:
         """
-        Draws an image with annotations overlaid.
-        :param image_token: The token of the image to be rendered.
+        Renders an image (sample_data), optionally with annotations overlaid.
+        :param sd_token_camera: The token of the sample_data to be rendered.
         :param with_annotations: Whether to draw all annotations.
+        :param with_category: Whether to include the category name at the top of a box.
         :param with_attributes: Whether to include attributes in the label tags.
-        :param box_tokens: List of bounding box annotation tokens. If given, only these annotations are drawn.
+        :param object_tokens: List of object annotation tokens. If given, only these annotations are drawn.
         :param surface_tokens: List of surface annotation tokens. If given, only these annotations are drawn.
```

Comment:
Same comment as for `object_tokens` argument; having a method (or showing in the tutorial how) to get a list of surface tokens belonging to the image would make this argument more useful to the user